### ISSUE TITLE
[Bug]: Error not thrown when using self-signed certificate with clientCertificates passed to Browser.newContext() and making requests with BrowserContext.request

### ISSUE THREAD

**Issue by oprysok** [microsoft/playwright#32973]
### Version

1.47.2

### Steps to reproduce

1. Clone reproduction [repo](https://github.com/oprysok/playwright-apirequestcontext-tls-issue.git)
2. Install dependencies: `pnpm i` or `npm i`
3. Run: `pnpm run test` or `npm run test`

When the tests start, Playwright launches a simple HTTPS server using a self-signed certificate. There are two tests, both of which send requests to the `/api/sample/` endpoint using `ApiRequestContext` with client certificate configurations.

In the first test, the client certificate configuration is passed to `Browser.newContext()`, and the request is made using the request property.
In the second test, the global `request` is used (`APIRequest.newContext()`).

### Expected behavior

Both test to pass. In each test we expect to get `apiRequestContext.get: self-signed certificate` error.

### Actual behavior

First test fails.

### Environment

```shell
System:
    OS: macOS 14.6
    CPU: (12) arm64 Apple M3 Pro
    Memory: 46.44 MB / 36.00 GB
  Binaries:
    Node: 20.17.0 - ~/.nvm/versions/node/v20.17.0/bin/node
    npm: 10.8.2 - ~/.nvm/versions/node/v20.17.0/bin/npm
    pnpm: 9.11.0 - ~/.nvm/versions/node/v20.17.0/bin/pnpm
  IDEs:
    VSCode: 1.93.1 - /opt/homebrew/bin/code
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: ^1.47.2 => 1.47.2
```

**Comment by mxschmitt** [comment]
It was actually a bug that it was working in one but not the other. (already fixed in our main branch and available [here](https://playwright.dev/docs/canary-releases)).

The expected outcome is that it throws in both cases, since you have a self-signed certificate. To fix this, you need to pass `ignoreHTTPSErrors: true` in your config/context.

**Comment by oprysok** [comment]
@mxschmitt Thanks for your prompt response! I tested it with the Canary version, and now both requests work consistently. I'll update the issue title and expected results to clarify—an error should be thrown in both cases unless a self-signed certificate is explicitly trusted.

**Comment by mxschmitt** [comment]
I don't follow - so if I clone your repository it fails on 1.47 but passes with Canary - so looks like we addressed your issue already? (fix will be included in the next v1.48)

**Comment by oprysok** [comment]
Correct

**Comment by mxschmitt** [comment]
Closing as per https://github.com/microsoft/playwright/issues/32973#issuecomment-2395593229 - should be fixed in v1.48 which we'll release soon.
